openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -48,16 +48,16 @@ def __init__(self, course_key, request, username=''):
             course_key,
         )
         self.effective_user = self.overview.effective_user
-        # We need to memoize `is_staff` _before_ we configure masquerade.
-        self.is_staff = has_access(self.effective_user, 'staff', self.overview).has_access
+        self.original_user_is_staff = has_access(request.user, 'staff', self.overview).has_access
         self.course_key = course_key
         self.enrollment_object = CourseEnrollment.get_enrollment(self.effective_user, self.course_key,
                                                                  select_related=['celebration'])
-        course_masquerade, _user = setup_masquerade(
+        course_masquerade, user = setup_masquerade(
             request,
             course_key,
-            staff_access=self.is_staff,
+            staff_access=self.original_user_is_staff,
         )
+        self.is_staff = has_access(user, 'staff', self.overview).has_access
```

Comment:
Are there any other consumers of this API about which we should be concerned with backward-compatibility?

Reply:
I'm fairly certain it's just the courseware MFE.

Reply:
Yes, I agree with David here. I believe both endpoints changed in this PR are solely consumed by the Learning MFE and were designed with that sole purpose in mind.